src/roles/httpd/tasks/main.yml

```diff
@@ -45,11 +36,25 @@
 - name: Configure foreman-ssl vhost
   ansible.builtin.template:
     src: foreman-ssl-vhost.conf.j2
-    dest: /etc/httpd/conf.d/foreman-ssl.conf
+    dest: "{{ httpd_conf_dir }}/foreman-ssl.conf"
     mode: "0644"
 
-- name: Start Apache httpd
-  ansible.builtin.service:
+- name: Disable welcome page in container
+  ansible.builtin.file:
+    path: "{{ httpd_conf_dir }}/welcome.conf"
+    state: absent
+
+- name: Run Apache as a container
+  containers.podman.podman_container:
     name: httpd
+    image: "{{ httpd_container_image }}"
     state: started
-    enabled: true
+    volumes:
+      - "{{ httpd_ssl_dir }}:/etc/httpd/ssl"
+      - "{{ httpd_conf_dir }}:/etc/httpd/conf.d"
```

Comment:
I think mapping the who configuration directory will overwrite other files and you're mixing host and container configs this way.